spec/integration/aws_auto_scaling_group_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'spec_helper'
+
+describe Chef::Resource::AwsAutoScalingGroup do
+  extend AWSSupport
+
+  when_the_chef_12_server 'exists', organization: 'foo', server_scope: :context do
+    with_aws 'When connected to AWS' do
+      # Select an amazon linux ami based upon region
+      ami = {
+        'eu-west-1' => 'ami-3850624c',
+        'eu-central-1' => 'ami-993383ea',
+        'us-east-1' => 'ami-e024bf89',
+        'us-west-1' => 'ami-951945d0',
+        'us-west-2' => 'ami-16fd7026',
+        'ap-northeast-1' => 'ami-dcfa4edd',
+        'ap-northeast-2' => 'ami-7308c51d',
+        'ap-southeast-1' => 'ami-74dda626',
+        'ap-southeast-2' => 'ami-b5990e8f',
+        'sa-east-1' => 'ami-3e3be423'
+      }[driver.aws_config.region]
+
+      aws_launch_configuration 'test_config' do
+        image ami
+        instance_type 't1.micro'
+      end
+
+      aws_sns_topic 'test_topic'
+
+      it "aws_auto_scaling_group 'test_group' creates an auto scaling group" do
+        expect_recipe {
+          aws_auto_scaling_group 'test_group' do
+            launch_configuration 'test_config'
+            availability_zones ["#{driver.aws_config.region}a"]
+            min_size 1
+            max_size 2
+          end
+        }.to create_an_aws_auto_scaling_group(
+          'test_group').and be_idempotent
+      end
+
+      it "aws_auto_scaling_group 'test_group_with_policy' creates an auto scaling group" do
+        expect_recipe {
+          aws_auto_scaling_group 'test_group_with_policy' do
+            launch_configuration 'test_config'
+            availability_zones ["#{driver.aws_config.region}a"]
+            min_size 1
+            max_size 2
+            notification_configurations [{
+              topic: driver.build_arn(service: 'sns', resource: 'test_topic'),
+              types: [
+                'autoscaling:EC2_INSTANCE_LAUNCH',
+                'autoscaling:EC2_INSTANCE_TERMINATE'
+              ]
+            }]
+            scaling_policies(
+              test_policy: {
+                adjustment_type: 'ChangeInCapacity',
```

Comment:
Same thinking here. Should we have some kind of validation on the scaling policies adjustment types?